scheduler/src/cook/plugins/job_submission_modifier.clj

```diff
@@ -0,0 +1,57 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+
+(ns cook.plugins.job-submission-modifier
+  (:require [clojure.tools.logging :as log]
+            [cook.config :as config]
+            [cook.plugins.definitions :refer [choose-pool-for-job modify-job JobRouter JobSubmissionModifier]]
+            [cook.plugins.util]
+            [mount.core :as mount]))
+
+(defn pool-name->effective-pool-name
+  "Given a pool name and job from a submission returns the effective pool name"
+  [pool-name-from-submission job]
+  (if-let [job-router (config/job-routing-pool-name? pool-name-from-submission)]
+    (choose-pool-for-job job-router job)
+    (or pool-name-from-submission (config/default-pool))))
+
+(defrecord IdentityJobSubmissionModifier []
+  JobSubmissionModifier
+  ; The IdentityJobSubmissionModifier doesn't make any changes to what users submit except
```

Comment:
Aaah. I see whats going on here.

You need to move JobRouter functionality into JobSubmissionModifier. That's OK. but you should note that in the docs for JobRouter. That JobRouter is only run if JobSubmissionModifier calls out to it.

Reply:
ooo good point.